examples/throughput/subscriber.c

```diff
@@ -165,9 +165,14 @@ static int do_take (dds_entity_t reader)
 
   /* Take samples and iterate through them */
 
+  info = dds_alloc (sizeof(dds_sample_info_t) * MAX_SAMPLES);
```

Comment:
From the point of view of performance and determinism it is better to avoid working with dynamic memory in the process of receiving messages. In this regard, the array on the stack looked preferable.

Reply:
Performance is not one of the goals of this example anymore. Actually this example is too complex and should be replaced by a much more simple example.

Reply:
There is a middle way here, too, which is to keep the array around. The samples are a global, there's no reason ``info`` should be treated differently.

That said, @BartP6703's comment that these examples are great at showing how _not_ to program is entirely correct. So I can't be bothered by a suboptimal code change that satisfies Coverity if that's merely a precursor to improving the examples altogether.